src/config.rs

```diff
@@ -0,0 +1,725 @@
+pub(crate) struct AgentConfig {
+    pub uuid: String,
+    pub ip: String,
+    pub port: u32,
+    pub contact_ip: Option<String>,
+    pub contact_port: Option<u32>,
+    pub registrar_ip: String,
+    pub registrar_port: u32,
+    pub enable_agent_mtls: bool,
+    pub keylime_dir: String,
+    pub server_key: Option<String>,
+    pub server_cert: Option<String>,
+    pub server_key_password: Option<String>,
+    pub trusted_client_ca: Option<String>,
+    pub enc_keyname: String,
+    pub dec_payload_file: String,
+    pub secure_size: String,
+    pub tpm_ownerpassword: Option<String>,
+    pub extract_payload_zip: bool,
+    pub enable_revocation_notifications: bool,
+    pub revocation_actions_dir: Option<String>,
+    pub revocation_notification_ip: Option<String>,
+    pub revocation_notification_port: Option<u32>,
+    pub revocation_cert: Option<String>,
+    pub revocation_actions: Option<String>,
+    pub payload_script: String,
+    pub enable_insecure_payload: bool,
+    pub allow_payload_revocation_actions: bool,
+    pub tpm_hash_alg: String,
+    pub tpm_encryption_alg: String,
+    pub tpm_signing_alg: String,
+    pub ek_handle: Option<String>,
+    pub run_as: Option<String>,
+    pub agent_data_path: Option<String>,
+}
+
+impl Default for AgentConfig {
+    fn default() -> Self {
+        // In case the process is executed by privileged user
+        let run_as = if permissions::get_euid() == 0 {
+            Some(DEFAULT_RUN_AS.to_string())
+        } else {
+            None
+        };
+
+        AgentConfig {
+            ip: DEFAULT_IP.to_string(),
+            port: DEFAULT_PORT,
+            registrar_ip: DEFAULT_REGISTRAR_IP.to_string(),
+            registrar_port: DEFAULT_REGISTRAR_PORT,
+            uuid: DEFAULT_UUID.to_string(),
+            contact_ip: Some(DEFAULT_CONTACT_IP.to_string()),
+            contact_port: Some(DEFAULT_CONTACT_PORT),
+            tpm_hash_alg: DEFAULT_TPM_HASH_ALG.to_string(),
+            tpm_encryption_alg: DEFAULT_TPM_ENCRYPTION_ALG.to_string(),
+            tpm_signing_alg: DEFAULT_TPM_SIGNING_ALG.to_string(),
+            agent_data_path: Some("default".to_string()),
+            enable_revocation_notifications:
+                DEFAULT_ENABLE_REVOCATION_NOTIFICATIONS,
+            revocation_cert: Some("default".to_string()),
+            revocation_notification_ip: Some(
+                DEFAULT_REVOCATION_NOTIFICATION_IP.to_string(),
+            ),
+            revocation_notification_port: Some(
+                DEFAULT_REVOCATION_NOTIFICATION_PORT,
+            ),
+            secure_size: DEFAULT_SECURE_SIZE.to_string(),
+            payload_script: DEFAULT_PAYLOAD_SCRIPT.to_string(),
+            dec_payload_file: DEFAULT_DEC_PAYLOAD_FILE.to_string(),
+            enc_keyname: DEFAULT_ENC_KEYNAME.to_string(),
+            extract_payload_zip: DEFAULT_EXTRACT_PAYLOAD_ZIP,
+            server_key: Some("default".to_string()),
+            server_key_password: Some(
```

Comment:
... because this put `Some("")` (there is a password, that is the empty string), that is different from `None` (no password was provided)

Reply:
The empty string is considered "no password". Check:
https://github.com/ansasaki/rust-keylime/blob/036ab2a392022641a8603fddb405b58dd36430b8/src/crypto.rs#L77-L101
and
https://github.com/ansasaki/rust-keylime/blob/036ab2a392022641a8603fddb405b58dd36430b8/src/crypto.rs#L54-L72